src/transformers/models/blip/modeling_blip.py

```diff
@@ -0,0 +1,1450 @@
+from .modeling_blip_text import BlipTextLMHeadModel, BlipTextModel
+
+
+logger = logging.get_logger(__name__)
+
+_CHECKPOINT_FOR_DOC = "Salesforce/blip-vqa-base"
+
+BLIP_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "Salesforce/blip-vqa-base",
+    "Salesforce/blip-vqa-base-capfit",
+    "Salesforce/blip-image-captioning-base",
+    "Salesforce/blip-image-captioning-large",
+    "Salesforce/blip-retrieval-coco-base",
+    "Salesforce/blip-retrieval-coco-large",
+    "Salesforce/blip-retrieval-flikr-base",
+    "Salesforce/blip-retrieval-flikr-large",
+    # See all BLIP models at https://huggingface.co/models?filter=blip
+]
+
+
+# Copied from transformers.models.clip.modeling_clip.contrastive_loss
+def contrastive_loss(logits: torch.Tensor) -> torch.Tensor:
+    return nn.functional.cross_entropy(logits, torch.arange(len(logits), device=logits.device))
+
+
+# Copied from transformers.models.clip.modeling_clip.clip_loss with clip->blip
+def blip_loss(similarity: torch.Tensor) -> torch.Tensor:
+    caption_loss = contrastive_loss(similarity)
+    image_loss = contrastive_loss(similarity.t())
+    return (caption_loss + image_loss) / 2.0
+
+
+@dataclass
+class BlipForConditionalGenerationModelOutput(ModelOutput):
+    """
+    Adapted from the base class for vision model's outputs that also contains image embeddings of the pooling of the
+    last hidden states. This class also adds the loss term from the text decoder.
+
+    Args:
+        loss (*optional*, *optional* returned when `labels` is provided, `torch.FloatTensor` of shape `(1,)`):
+            Languge modeling loss from the text decoder.
+        image_embeds (*torch.FloatTensor* of shape *(batch_size, output_dim)* *optional* returned when model is initialized with *with_projection=True*):
+            The image embeddings obtained by applying the projection layer to the pooler_output.
+        last_hidden_state (*torch.FloatTensor* of shape *(batch_size, sequence_length, hidden_size)*):
+            Sequence of hidden-states at the output of the last layer of the model.
+        hidden_states (*tuple(torch.FloatTensor)*, *optional*, returned when *output_hidden_states=True* is passed or when *config.output_hidden_states=True*):
+            Tuple of *torch.FloatTensor* (one for the output of the embeddings, if the model has an embedding layer, +
+            one for the output of each layer) of shape *(batch_size, sequence_length, hidden_size)*.
+
+            Hidden-states of the model at the output of each layer plus the optional initial embedding outputs.
+        attentions (*tuple(torch.FloatTensor)*, *optional*, returned when *output_attentions=True* is passed or when *config.output_attentions=True*):
+            Tuple of *torch.FloatTensor* (one for each layer) of shape *(batch_size, num_heads, sequence_length,
+            sequence_length)*.
+
+            Attentions weights after the attention softmax, used to compute the weighted average in the self-attention
+            heads.
```

Comment:
Lost of code blocks are in-between stars instead of backtiks.

Reply:
Fixed in 829f2b3 ! (GitHub copilot is not that good at the end)